app/models/profile.rb

```diff
@@ -34,7 +21,38 @@ def activate
     end
   end
 
-  def verified?
-    verified_at.present?
+  def plain_pii
+    @_plain_pii ||= Pii::Attributes.new
+  end
+
+  def decrypt_pii(password)
+    Pii::Attributes.new_from_encrypted(encrypted_pii, password, salt)
+  end
+
+  def encrypt_pii(password, pii = plain_pii)
+    ssn = pii.ssn
+    self.ssn_signature = Pii::Fingerprinter.fingerprint(ssn) if ssn
+    self.encrypted_pii = pii.encrypted(password, salt)
+  end
+
+  def method_missing(method_sym, *arguments, &block)
+    attr_name_sym = method_sym.to_s.gsub(/=\z/, '').to_sym
+    if plain_pii.members.include?(attr_name_sym)
+      return plain_pii[attr_name_sym] if arguments.empty?
+      plain_pii[attr_name_sym] = arguments.first
```

Comment:
Again, I'd be really happy with just writing `profile.plain_pii.foobar` everywhere since it makes for a really clear relation, but it may be easier to refactor the code with this delegation layer? Would love to get other folk's opinions.

(Also teeny nit, structs have a class-level `.members` so you could do `Pii::Attributes.members` without the `.new`)